packages/core/src/transport/failedSendBeacon.ts

```diff
@@ -0,0 +1,43 @@
+import { isExperimentalFeatureEnabled } from '../domain/configuration'
+import { addMonitoringMessage, monitor } from '../domain/internalMonitoring'
+import { generateUUID, startsWith } from '../tools/utils'
+
+// replaced at build time
+declare const __BUILD_ENV__SDK_VERSION__: string
+
+export const LOCAL_STORAGE_KEY = 'datadog-browser-sdk-failed-send-beacon'
+
+export function startFlushFailedSendBeacons() {
+  if (!isExperimentalFeatureEnabled('lower-batch-size')) return
+
+  setTimeout(monitor(flushFailedSendBeacon))
+}
+
+export function addFailedSendBeacon(endpointType: string, size: number, reason?: string) {
+  if (!isExperimentalFeatureEnabled('lower-batch-size')) return
+
+  window.localStorage.setItem(
+    `${LOCAL_STORAGE_KEY}-${generateUUID()}`,
+    JSON.stringify({
+      reason,
+      endpointType,
+      version: __BUILD_ENV__SDK_VERSION__,
+      connection: navigator.connection ? (navigator.connection as any).effectiveType : undefined,
+      onLine: navigator.onLine,
+      size,
+    })
+  )
+}
```

Comment:
I made so that we only buffer logs in local storage on before_unload and visibility_hidden. I used both event because before_unload isn't reliable in all case (like mobile). Let me know what you think